src/magic/castSpell.cpp

```diff
@@ -110,6 +110,39 @@ void castSpellInit(Uint32 caster_uid, spell_t* spell)
 		return;
 	}
 
+    // Check to make sure the Caster is not swimming
```

Comment:
There's a lot of conditions here which are duplicated (not only in here but in `actplayer.cpp`, `acthudweapon.cpp`, which should probably be factored out to reduce the duplication and the amount of code that needs to be changed should additional conditions for swimming be added.